app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,73 @@
+class TweetsController < Sinatra::Base
+
+  include Helper
+
+  configure do
+    set :public_folder, 'public'
+    set :views, "app/views/tweets"
+    enable :sessions
+    set :session_secret, "what a wonderful secret this is"
+    use Rack::Flash
+  end
+
+  get "/tweets" do
+    @user = current_user
+    @tweets = Tweet.all
+    erb :tweets
+  end
+
+  get "/tweets/new" do
+    erb :new
+  end
+
+  post "/tweets" do
+    if params[:tweet][:content].empty?
+      flash[:message] = "Tweet contents cannot be empty!"
+      redirect to("/tweets/new")
+    else
+      current_user.tweets.create(params[:tweet])
+      redirect to("/tweets")
+    end
+  end
+
+  get "/tweets/:id" do
+    @user = current_user
+    @tweet = Tweet.find(params[:id])
+    erb :show_tweet
+  end
+
+  get "/tweets/:id/edit" do
+    @tweet = Tweet.find(params[:id])
+    if @tweet.user == current_user
+      erb :edit_tweet
+    else
+      flash[:message] = "This is not your tweet to edit!"
+      redirect to("/tweets/#{@tweet.id}")
+    end
+  end
+
+  patch "/tweets/:id" do
+    tweet = Tweet.find(params[:id])
+    if params[:tweet][:content].empty?
+      flash[:message] = "Tweet contents cannot be empty!"
+      redirect to("/tweets/#{tweet.id}/edit")
+    elsif tweet.user == current_user
+      tweet.update(params[:tweet])
+    else
+      flash[:message] = "This is not your tweet to edit!"
+    end
+    redirect to("/tweets/#{tweet.id}")
+  end
+
+  delete "/tweets/:id/delete" do
+    tweet = Tweet.find(params[:id])
+    if tweet.user == current_user
+      tweet.destroy
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.